### PR TITLE
Fix the bug where Get- cmdlets does not include the result of the nex…

### PR DESCRIPTION
**Pull request by hyonholee**
…t page link.

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by cormacpayne** [comment]
@hyonholee ping

**Comment by hyonholee** [comment]
I will update the test soon.  Thanks.

**Comment by hyonholee** [comment]
@markcowl  I added new test for nextlink.  Thanks.

File: src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 [hunk 1]
```diff
@@ -580,3 +580,56 @@ function Test-VirtualMachineScaleSetLB
...
+        $vmss = New-AzureRmVmssConfig -Location $loc -SkuCapacity $vmss_number -SkuName 'Standard_A0' -UpgradePolicyMode 'Automatic' -NetworkInterfaceConfiguration $netCfg -Overprovision $false -SinglePlacementGroup $false `
+            | Add-AzureRmVmssNetworkInterfaceConfiguration -Name 'test' -Primary $true -IPConfiguration $ipCfg `
+            | Set-AzureRmVmssOSProfile -ComputerNamePrefix 'test' -AdminUsername $adminUsername -AdminPassword $adminPassword `
+            | Set-AzureRmVmssStorageProfile -OsDiskCreateOption 'FromImage' -OsDiskCaching 'None' `
+            -ImageReferenceOffer $imgRef.Offer -ImageReferenceSku $imgRef.Skus -ImageReferenceVersion $imgRef.Version -ImageReferencePublisher $imgRef.PublisherName;
+
+        $result = New-AzureRmVmss -ResourceGroupName $rgname -Name $vmssName -VirtualMachineScaleSet $vmss;
+
+        $vmssVmResult = Get-AzureRmVmssVM -ResourceGroupName $rgname -VMScaleSetName $vmssName;
+        Assert-AreEqual $vmss_number $vmssVmResult.Count
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1]: @hyonholee what would `$vmssVmResult.Count` have resolved to before this fix?

Comment (hyonholee) [inline src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1]: In this test, VM scale set creates 180 vms.  If the number of VMs in VM scale set is greater than 120, Get operation returns "nextLink".  In such a case, Get-AzureRmVmssVM cmdlet calls the next link and is supposed to include all of the VMs returned by the next link to the output.  However, there was a bug that those VMs were not included in the output, and Get-AzureRmVmssVM returns only 120 VMs.    So, in this test, it compares the Get-AzureRmVmssVM returns the correct number of VMs even the VM scale set has more than 120 VMs.

**Comment by cormacpayne** [comment]
LGTM

@markcowl any other comments?

File: src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineTests.cs [hunk 2]
```diff
@@ -157,7 +157,7 @@ public void TestGetVMSizeFromAllLocations()
             ComputeTestController.NewInstance.RunPsTest("Test-GetVMSizeFromAllLocations");
         }
 
-        [Fact(Skip = "TODO: OOM issue when writing the result")]
+        [Fact]
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/VirtualMachineTests.cs]: Please make this a check-in test

**Comment by cormacpayne** [comment]
@hyonholee ping

**Comment by cormacpayne** [comment]
@markcowl ping

File: src/ResourceManager/Compute/ChangeLog.md [hunk 3]
```diff
@@ -19,6 +19,8 @@
 -->
 ## Current Release
 
+* Fix the bug of Get cmdlets, where the output did not include all of the results from next page links, if the result contains more than 120 items.
```
Comment (markcowl) [inline src/ResourceManager/Compute/ChangeLog.md]: How about:  Fixed bug to allow Get- cmdlets to return more than 120 results?  Does this apply to all compute get cmdlets, or just for container services or vmss?

**Comment by markcowl** [comment]
LGTM once the build passes

**Pull request merged**